backend/routes/fellows.js

```diff
@@ -0,0 +1,134 @@
+/*
+ANIME BENSALEM, BRIAHANA MAUGÉ, JOSEPH P. PASAOA
+FELLOWS Route Handler | Capstone App (Pursuit Volunteer Mgr)
+*/
+
+
+/* MODULE INITS */
+const express = require('express');
+const router = express.Router();
+
+const handleError = require('../helpers/handleError');
+const processInput = require('../helpers/processInput');
+const queries = require('../queries/fellows');
+
+
+/* MIDDLEWARE FUNCTIONS */
+const readAllFellows = async (req, res, next) => {
+  try {
+    const allFellows = await queries.getAllFellows();
+    res.status(200);
+    res.json({
+        status: "success",
+        message: "all fellows retrieved",
+        payload: allFellows
+    });
+  } catch (err) {
+    handleError(err, req, res, next);
+  }
+};
+
+const readFellow = async (req, res, next) => {
+  try {
+    const fellowId = processInput(req.params.fellow_id, res, "idNum", "fellow id");
```

Comment:
I think the way it's set up now, if you have a non-number into req.params.fellow_id then processInput will send a response but the code will continue to run (const fellowById = await queries.getFellowById(fellowId)) then this will throw an error and `catch` will handle that but it will send another response causing a backend error `you can't send a response after the response was sent (or something similar, I don't remember the error) `

Reply:
@AminesCodes this was the main reason my original system was based on throws because in sending the responses from processInput instead the code continues to run like you mentioned above. i was able to spend an hour trying to make this work but i ended up slightly reverting back to my throw > funneled catch system which worked before and didn't have the split response errors